nextcord/channel.py

```diff
@@ -577,8 +583,8 @@ async def create_webhook(self, *, name: str, avatar: Optional[bytes] = None, rea
 
         from .webhook import Webhook
 
-        if avatar is not None:
-            avatar = utils._bytes_to_base64_data(avatar)  # type: ignore
+        avatar = avatar if avatar is None else utils._bytes_to_base64_data(
+            avatar if isinstance(avatar, bytes) else await avatar.read())
```

Comment:
It's best to avoid having multiple conditional operations in a single statement. This makes the code more difficult to comprehend.

I'd recommend spreading it out across multiple lines, or use a `utils` function to abstract the logic.